HW_15/task1.py

```diff
@@ -0,0 +1,185 @@
+import json, re
+from time import localtime, strftime
+
+# # Make new .json with testing data. Run once.
+# phonebook_dict = {
+#     "+11111111": ["Tom", "Jones"],
+#     "+33333333": ["Bob", "Marley"],
+#     "+55555555": ["Alice", "Cooper"],
+#     "+44444444": ["Tom", "Jones"],
+#     "+77777777": ["Tom", "Cruise"],
+#     "+88888888": ["Tom and Jerry", "Pets"],
+#     "101":["q", "w"]
+# }
+#
+# with open('phonebook_dict.json', "x") as phonebook:
+#     json.dump(phonebook_dict, phonebook)
+
+phonebook_dict = {}
+json_string = {}
+
+with open('phonebook_dict.json', "rt") as phonebook_json:
+    phonebook_dict = json.load(phonebook_json)
+
+# ===== Decorator section
+def definition_running_time(func):
+    def wraper(*args):
+        func(*args)
+        with open('definition_running_time.log', "a") as definition_log_file:
+            definition_log_file.write(f"""{func.__name__} run at {strftime("%H:%M:%S", localtime())}\n""")
+    return wraper
+
+
+# ===== Custom exception
+class MyCustomException(Exception):
+    def __init__(self, message, error_code):
+        super().__init__(message)
+        with open('my_custom_exeption.log', "a") as exeption_log_file:
+            exeption_log_file.write(f"""{(message)} with code {error_code} excepted at {strftime("%H:%M:%S", localtime())}\n""")
+        self.error_code = error_code
+
+    @definition_running_time
+    def get_error_code(self):
+        return self.error_code
+
+
+# ===== Definition section
+@definition_running_time
+def write_file(phonebook_dict):
+    with open('phonebook_dict.json', "w") as phonebook:
+        json.dump(phonebook_dict, phonebook)
+
+@definition_running_time
+def stats_command():
+    print(f"Count of items: {len(phonebook_dict)}")
+    return
+
+# @definition_running_time   #  Make validation is always False. Need to remake a "definition_running_time" definition (((
+def phone_validation(new_phone_num) -> bool:
+    return bool(re.search(valid_phone_pattern, new_phone_num))
+
+@definition_running_time
+def add_command(user_input):
+    user_input_split = user_input.split()
+
+    """ Removing non-digit symbol from phone number (google thx). 
+    Several annotation:
+    1. Phone number always will be written in phone book without beginning '+'
+    2. 380501234567 != 0501234567 
+    """
```

Comment:
nit: better to write comment using `#` iside code and `"""` only describing functions and classes